frontend/src/bundles/common/components/calendar/calendar.tsx

```diff
@@ -0,0 +1,186 @@
+import clsx from 'clsx';
+import { type ChangeEvent,useCallback } from 'react';
+import { useEffect, useRef, useState } from 'react';
+import { Calendar as ReactCalendar } from 'react-calendar';
+import { type Control, type FieldPath, type FieldValues } from 'react-hook-form';
+
+import ArrowImage from '~/assets/img/arrow.svg';
+
+import { useAppForm, useFormController } from '../../hooks/hooks';
+import { type CalendarDate, type CalendarMonth } from '../../types/types';
+import { Toggle } from '../components';
+import { CalendarMonths, DEFAULT_DATE_PAYLOAD, monthRegex,yearRegex } from './constants/calendar.constants';
+import styles from './styles.module.scss';
+
+type Properties<T extends FieldValues> = {
+    control: Control<T, null>;
+    name: FieldPath<T>;
+    showPresent?: boolean;
+    className?: string
+};
+
+const Calendar = <T extends FieldValues>({
+    showPresent = false,
```

Comment:
Looks like a callback, try to come up with better name

Reply:
done